### PR TITLE
Add controller nrtmv4 file

### PR DESCRIPTION
**Pull request by maggarwal13**

### CODE REVIEW

File: whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+
+    public static final String SOURCE_LINK_PAGE = "<html><header><title>NRTM Version 4</title></header><body>%s<body></html>";
+    private final SnapshotSourceAwareFileRepository snapshotSourceAwareFileRepository;
+    private final DeltaSourceAwareFileRepository deltaSourceAwareFileRepository;
+    private final NotificationFileSourceAwareDao notificationFileSourceAwareDao;
+    private final SourceRepository sourceRepository;
+    final String nrtmUrl;
+
+    @Autowired
+    public NrtmClientService(@Value("${nrtm.baseUrl}") final String nrtmUrl,
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java:39]: Should this property be mandatory? We need a way to turn NRTMv4 off in certain environments.

File: whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        }
+
+        final String filenameExt  = filenameWithExtension(fileName);
+        if(fileName.startsWith(NrtmDocumentType.DELTA.getFileNamePrefix())) {
+            return deltaSourceAwareFileRepository.getByFileName(filenameExt)
+                    .map( delta -> getResponse(delta.payload(), delta.hash(), filenameExt))
+                    .orElseThrow(() -> new NotFoundException("Requested Delta file does not exists"));
+        }
+
+        throw new BadRequestException("Invalid Nrtm filename");
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java:97]: Return 404 Not Found instead?

File: whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    }
+
+    private Response getResponse(final String payload, final String filename) {
+        return addCommonHeader(Response.ok(payload), payload.length(), filename, MediaType.APPLICATION_JSON)
+                .build();
+    }
+
+    private Response.ResponseBuilder addCommonHeader(final Response.ResponseBuilder builder, final int length, final String filename, final String type) {
+        return builder.header(HttpHeaders.CONTENT_LENGTH, length)
+                .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=" + filename)
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java:133]: Don't force download as a file

File: whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java (hunk 2 above)
Comment (maggarwal13) [reply whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java:97]: Not really it is invalid file now , as it does not match to any of snapshot/delta or notification file syntax

File: whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java (hunk 1 above)
Comment (maggarwal13) [reply whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java:39]: I will make it optional

File: whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+                             final SnapshotSourceAwareFileRepository snapshotSourceAwareFileRepository,
+                             final DeltaSourceAwareFileRepository deltaSourceAwareFileRepository) {
+        this.snapshotSourceAwareFileRepository = snapshotSourceAwareFileRepository;
+        this.deltaSourceAwareFileRepository = deltaSourceAwareFileRepository;
+        this.notificationFileSourceAwareDao = notificationFileSourceAwareDao;
+        this.sourceRepository = sourceRepository;
+        this.nrtmUrl = nrtmUrl;
+    }
+
+    @GET
```
Comment (eshryane) [inline whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java:51]: Does this method need a `@Path("/")` ?

Comment (maggarwal13) [reply whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java:51]: No, there is integration tests to prove this. As we have a mapping on root level

Comment (eshryane) [reply whois-api/src/main/java/net/ripe/db/whois/api/nrtm4/NrtmClientService.java:51]: OK! Thanks

**Review by eshryane — APPROVED** [review]

**Pull request merged**